support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx

```diff
@@ -206,6 +206,25 @@ function OneTimeCheckoutComponent({
 
 	const formRef = useRef<HTMLFormElement>(null);
 
+	const displayValidity = (
```

Comment:
Is this clearer than the code it encompasses? The code in this method is standard HTML apis, whereas this is a new custom one. 

I do think it looks a little neater in the hander methods, but obfuscates what's actually happening.

I don't have a preference really, just asking.

Reply:
Think mainly was reducing duplication of code as its content was repeated  3 times.

Reply:
Repetition of code is not necessarily a bad thing - sometimes things happen multiple times. 

There are multiple reasons to abstract something
- encapsulating good practise e.g. `POST`ing to an auth endpoint with creds
- Simplification and understandability - Avoiding having a future person work out how to do something complicated e.g. potentially this?


There's more. Just wondering - from your perspective - what you are achieving with this abstraction?

Reply:
To be clear though - if we think this makes it more readable, that's a good reason, especially that's it's encapsulated to this package and not leaky.